it/src/main/java/org/corfudb/universe/universe/docker/FakeDns.java

```diff
@@ -0,0 +1,187 @@
+            // Create a proxy instance to set on the InetAddress field which will handle
+            // all NameService calls.
+            Object proxy = Proxy.newProxyInstance(
+                    nameServiceInterface.getClassLoader(),
+                    new Class<?>[]{nameServiceInterface},
+                    new NameServiceListener(fallbackNameService)
+            );
+            field.setAccessible(true);
+            // Java 8 or earlier takes a list of NameServices
+            field.set(InetAddress.class, Arrays.asList(proxy));
+        }
+    }
+
+    /**
+     * The NameService in all versions of Java has the same interface, so we
+     * can use the same InvocationHandler as our proxy instance for both
+     * java.net.InetAddress$NameService and sun.net.spi.nameservice.NameService.
+     */
+    private class NameServiceListener implements InvocationHandler {
+
+        private final Object fallbackNameService;
+
+        // Creates a NameServiceListener with a NameService implementation to
+        // fallback to. The parameter is untyped so we can handle the NameService
+        // type in all versions of Java with reflection.
+        NameServiceListener(Object fallbackNameService) {
+            this.fallbackNameService = fallbackNameService;
+        }
+
+        private InetAddress[] lookupAllHostAddr(String host) throws UnknownHostException {
+            InetAddress inetAddress;
+            synchronized (FakeDns.this) {
+                inetAddress = forwardResolutions.get(host);
+            }
+            if (inetAddress != null) {
+                return new InetAddress[]{inetAddress};
+            }
+
+            try {
+                Method method = fallbackNameService.getClass().getDeclaredMethod("lookupAllHostAddr", String.class);
+                method.setAccessible(true);
+                return (InetAddress[]) method.invoke(fallbackNameService, host);
+            } catch (ReflectiveOperationException | NoSuchElementException | SecurityException e) {
+                Throwables.propagateIfPossible(e.getCause(), UnknownHostException.class);
+                throw new AssertionError("unexpected reflection issue", e);
+            }
+        }
+
+        private String getHostByAddr(byte[] addr) throws UnknownHostException {
+            if (addr[0] == 127) {
+                return InetAddresses.toAddrString(InetAddress.getByAddress(addr));
+            }
+
+            String hostname;
+            synchronized (FakeDns.this) {
+                hostname = reverseResolutions.get(InetAddress.getByAddress(addr));
+            }
+            if (hostname != null) {
+                return hostname;
+            }
+
+            try {
+                Method method = fallbackNameService
+                        .getClass()
+                        .getDeclaredMethod("getHostByAddr", byte[].class);
+
+                method.setAccessible(true);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make sure that this accessibility update is safe here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3011)